nni/contrib/compression/base/target_space.py

```diff
@@ -0,0 +1,372 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+from __future__ import annotations
+
+from collections import abc
+from copy import deepcopy
+from enum import Enum
+from typing import Any, Dict, List, Literal, Tuple
+
+import torch
+from torch import Tensor
+
+from nni.compression.pytorch.utils.scaling import Scaling
+
+
+class TargetType(Enum):
+    INPUT = 'input'
+    OUTPUT = 'output'
+    PARAMETER = 'parameter'
+
+
+class TargetSpace:
```

Comment:
What's the purpose of a TargetSpace?

Reply:
add class doc str
```
    """
    TargetSpace is a container to handle all compression information related to a specific compression target,
    note that it has the ability to register information in the wrapper,
    this because in most comopression cases need to register tensor (mask, scale, zero point) as buffer/parameter in the model.

    Parameters
    ----------
    wrapper
        The wrapper of a module.
    target_name
        If the target is a parameter, the name is the attr name in the module, i.e., 'weight', 'bias'.
        If the target is the module input/output, the name should have prefix '_input_'/'_output_',
        then follow a number or name, i.e., '_input_0', '_output_hidden_state',
        number is the position of this input/output in all inputs/outputs,
        name is the input argument name or outputs dict key.
    target_type
        'input' or 'output' or 'parameter'.
    setting
        The compression setting, i.e., {'sparse_ratio': 0.1}.
    """
```